lib/omb-prompt-base.sh

```diff
@@ -538,6 +538,15 @@ function _omb_prompt_get_python_env {
   [[ $python_env ]]
 }
 
+## @fn _omb_prompt_get_spack_env
+##   @var[out] spack_env
+##   @exit
+function _omb_prompt_get_spack_env {
+  spack_env=
+  [[ ${SPACK_ENV-} ]] || return 1
+  _omb_prompt_format spack_env "[$(basename "$SPACK_ENV")] " OMB_PROMPT_SPACK_ENV:SPACK_ENV_THEME_PROMPT
```

Comment:
```suggestion
  _omb_prompt_format spack_env "$(basename "$SPACK_ENV")" OMB_PROMPT_SPACK_ENV
```

- Any formatting such as enclosing the value within "[...]" should be performed through the corresponding formatting variable, `OMB_PROMPT_SPACE_ENV_FORMAT='[%s] '`. Please add this default value around line 83.
- The name after the colon in the last argument of `_omb_prompt_format` is for the deprecated variable names. In this case, `:SPACK_ENV_THEME_PROMPT` would imply that it also checks the *deprecated* variables `SPACK_ENV_THEME_PROMPT_PREFIX` and `SPACK_ENV_THEME_PROMPT_SUFFIX`. However, we shouldn't need to support these variables since we have never provided a feature to use `SPACK_ENV_THEME_PROMPT_PREFIX` and `SPACK_ENV_THEME_PROMPT_SUFFIX`. We can simply drop it.